test/DynamicExpresso.UnitTest/ConstructorTest.cs

```diff
@@ -179,6 +179,84 @@ public void Array_multi_dimension_constructor()
 			Assert.Throws<ParseException>(() => target.Parse("new int[,] { { 1 }, { 2 } }"));
 		}
 
+		[Test]
+		public void Ctor_NewDictionaryWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.Dictionary<,>));
+			var l = target.Eval<System.Collections.Generic.Dictionary<int, string>>("new Dictionary<int, string>(){{1, \"1\"}, {2, \"2\"}, {3, \"3\"}, {4, \"4\"}, {5, \"5\"}}");
+			Assert.AreEqual(5, l.Count);
+			for (int i = 0; i < l.Count; ++i)
+			{
+				Assert.AreEqual(i + 1 + "", l[i + 1]);
+			}
+		}
+
+		[Test]
+		public void Ctor_NewMyClassWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			var l = target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{\"6\" },7	}.Add(true)");
+			Assert.AreEqual(5, l.MyArr.Length);
+			for (int i = 0; i < l.MyArr.Length; ++i)
+			{
+				Assert.AreEqual(i + 1, l.MyArr[i]);
+			}
+			Assert.AreEqual("6", l.StrProp);
+			Assert.AreEqual(7, l.IntField);
+		}
+
+
+		[Test]
+		public void Ctor_NewMyClassWithCross()
+		{
+			string StrProp = "";
+			var mc = new MyClassAdder()
+			{
+				{1,2,3 },
+				{StrProp = "6" }
+			};
+			if (StrProp == "6")
+			{
+
+			}
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7}", new Parameter("StrProp", "0")));
+			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},StrProp = \"6\",7}", new Parameter("StrProp", "0")));
+			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},string.Empty, 7}"));
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7	}"));
+		}
+		[Test]
+		public void Ctor_NewListWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.List<>));
+			var intList = target.Eval<System.Collections.Generic.List<int>>("new List<int>(){1, 2, 3, 4, 5}");
```

Comment:
Can you add a test with a property access? For example, replace 5 with `int.MaxValue`, or a parameter? To ensure you can use any expression (I saw the next test uses a method call, but one more test won't hurt).

Reply:
@holdenmai I think this is the last missing test. Or it is already done?

Reply:
the Ctor_NewListWithItems has a lot of different tests calling out to different "constants", both direct and referencing out to int.MaxValue.

and Ctor_NewListWithString also contains references to using parameters.